src/test/java/org/ohdsi/analysis/versioning/SemverUtilsTest.java

```diff
@@ -0,0 +1,103 @@
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.ohdsi.analysis.versioning.SemverUtils.getRangesIntersection;
+
+public class SemverUtilsTest {
+
+
+    @Test
+    public void testGetRangesIntersectionSingle() {
+        String intersection = getRangesIntersection(singletonList(">=5.2"));
+        assertEquals(">=5.2", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionSimple() {
+        String intersection = getRangesIntersection(asList(">=5.0.0", ">=3.0.0"));
+        assertEquals(">=5.0.0", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionNoIntersection() {
+        String intersection = getRangesIntersection(asList(">=5.0.0", "3.0.0"));
+        assertNull(intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionWithSame1() {
+        String intersection = getRangesIntersection(asList(">=5.0.0", "5.0.0"));
+        assertEquals("5.0.0", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionWithSame2() {
+        String intersection = getRangesIntersection(asList(">=5.0.0", ">5.0.0"));
+        assertEquals(">5.0.0", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionWithSame3() {
+        String intersection = getRangesIntersection(asList("<=5.0.0", "<5.0.0"));
+        assertEquals("<5.0.0", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionWithSame4() {
+        String intersection = getRangesIntersection(asList(">=5.0.0", "<=5.0.0"));
+        assertEquals("5.0.0", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionWithRange1() {
+        String intersection = getRangesIntersection(asList(">=3.0.0", "<=5.0.0"));
+        assertEquals("3.0.0 - 5.0.0", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionWithRange2() {
+        String intersection = getRangesIntersection(asList("4 - 5", "<=5.0.0"));
+        assertEquals("4.0.0 - 5.0.0", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionWithRangeHack() {
+        String intersection = getRangesIntersection(asList("4.0 - 5.0", "<5"));
+        assertEquals("4.0.0 - 4.99.99", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionWithTwoRanges() {
+        String intersection = getRangesIntersection(asList("2.3.0 - 4.5.0", "3.0.0 - 5.3.0", "*"));
+        assertEquals("3.0.0 - 4.5.0", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionWithAstrix() {
+        String intersection = getRangesIntersection(asList("*", "<=5.0.0"));
+        assertEquals("<=5.0.0", intersection);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testGetRangesIntersectionWithEmptyList() {
+        getRangesIntersection(emptyList());
+    }
```

Comment:
There is another test that we should add that should result in an exception, which is the case where the range intersection results in nothing (i'm not sure if this should be excpetion or if it shoudl return null and if we need a cdm range somewhere we check for null):

```
getRangesIntersection(asList("<5",">6"));  /// should throw exception because this is a contradition or return NULL because there is no intersection.
```

Reply:
This kind of result returned null, and the `CdmVersionUtils.java` then threw an exception. With a null as result the contract stays in tact. I have a test `testGetRangesIntersectionNoIntersection` for this already, I've added another one that is slightly different but also has no intersection.

Reply:
That does make sense, I would however make it a separate issue, my intention with this change is to keep the functionality as much as it was and not change the behaviour in unexpected ways. Somebody downstream my be depending on that current hard failure to make sure their work is correct...

Reply:
Sounds good.